R/utilities-qualification.R

```diff
@@ -0,0 +1,235 @@
+    configurationPlan$copySectionContent(sectionId, logFolder = logFolder)
+  }
+  # Inputs
+  for (input in configurationPlan$inputs) {
+    configurationPlan$copyInput(input, logFolder = logFolder)
+  }
+  return(appendices)
+}
+
+
+
+#' @title getOutputsTimeProfile
+#' @description Get a vector of output paths that are to be used in a time profile plot
+#' @param plot is a descriptor of the plot that is read from the `ConfigurationPlan` that is generated by the configuration plan json
+#' @return A vector of output paths that are to be used in the time profile plot descriptor `plot`
+getOutputsTimeProfile <- function(plot) {
+  validateIsIncluded(values = "Plot", parentValues = names(plot), nullAllowed = TRUE)
+  validateIsIncluded(values = "Curves", parentValues = names(plot[["Plot"]]), nullAllowed = FALSE)
+
+  paths <- NULL
+  for (curve in plot$Plot$Curves) {
+    validateIsString(object = curve$Y)
+    if (ospsuite::toPathArray(curve$Y)[2] == "ObservedData") {
+      next
+    }
+    paths <- c(paths, ospsuite::toPathString(tail(ospsuite::toPathArray(curve$Y), -1)))
+  }
+  return(unique(paths))
+}
+
+
+#' @title getOutputsGOFMergedPlotsPlot
+#' @description Get a vector of output paths that are to be used in a GOF merged plot
+#' @param plot is a descriptor of the plot that is read from the `ConfigurationPlan` that is generated by the configuration plan json
+#' @return A vector of output paths that are to be used in the GOF merged plot descriptor `plot`
+getOutputsGOFMergedPlotsPlot <- function(plot) {
+  validateIsIncluded(values = "Groups", parentValues = names(plot), nullAllowed = TRUE)
+  paths <- NULL
+  for (group in plot$Groups) {
+    validateIsIncluded(values = "OutputMappings", parentValues = names(group), nullAllowed = TRUE)
+    for (outputMapping in group$OutputMappings) {
+      validateIsIncluded(values = "Output", parentValues = names(outputMapping), nullAllowed = TRUE)
+      validateIsString(object = outputMapping$Output)
+      print(outputMapping$Output)
+      paths <- c(paths, outputMapping$Output)
+    }
+  }
+  return(unique(paths))
+}
+
+#' @title getScheduleOfPlotsOfType
+#' @description Get a dataframe of plots of the plot type `plotType` to be generated from the `ConfigurationPlan` object that is generated by the configuration plan json
+#' @param pltList is a list of plots that is to be generated, all of the plot type `plotType`
+#' @param plotType describes the type of the plots described in `pltList`
+#' @return A dataframe describing the the the order of each plot among its plot type `plotType`, and the section id of each plot
+getScheduleOfPlotsOfType <- function(pltList, plotType) {
+  validateIsString(object = plotType)
+  schedule <- NULL
+  plotNumber <- 0
+  for (plt in pltList) {
+    plotNumber <- plotNumber + 1
+    schedule <- rbind.data.frame(schedule, data.frame(plotType = plotType, plotNumber = plotNumber, sectionId = plt$SectionId))
+  }
+  return(schedule)
+}
+

```

Comment:
@abdullahhamadeh , can you clarify the purpose of this function with a practical example ?
I am not sure why the plotNumber may be needed

And just to be consistent with the nomenclature you used, can you rename pltList into something like plotList or listOfPlots

Reply:
@pchelle This function is still WIP.  At one point I thought it would be good to have a dataframe that maps each plot a simulation set, and I started adding fields in the `schedule` dataframe that indicate 'project' and 'simulation'.  But there will be plots that are associated with more than one simulation set (see eg GOF plots and comparison plots in config plan), so the `schedule` dataframe may not be adequate on its own.